src/index.ts

```diff
@@ -1,48 +1,98 @@
 import Serverless = require("serverless");
 import { Options } from "serverless";
 import { Server } from "http";
+import { camelCase, paramCase as kebabCase } from "change-case";
 
 
 class ServerlessConventions {
      serverless: Serverless;
      hooks: { [key: string]: Function }
      config: any
+     errors: Array<string>
 
      constructor(serverless: Serverless, options: Options) {
           this.serverless = serverless;
-          this.config = ''
+          this.config = '';
+          this.errors = [];
 
           this.hooks = {
                'before:package:compileFunctions': this.initialize.bind(this)
           };
      }
 
      initialize() {
-          this.serverless.cli.log('Starting function handler check');
+          // Check that all the conventions are followed and build a list of errors
+          this.checkHandlerName();
+          this.checkFunctionName();
+          this.checkHandlerNameMatchesFunction();
+
+          // If there were errors detected throw them all
+          if (this.errors.length !== 0) {
+               throw Error(this.errors.join('\r\n'));
+          }
+          
+          this.serverless.cli.log('Function check complete! No errors were found.');
+     }
```

Comment:
I think it would be good to print out the errors (bonus points for RED colour) and then throw one final error to break the build. I.e `Error('Conventions validation failed')`.